### DISCUSSION TITLE
Iroh as a library?

### DISCUSSION THREAD

**Discussion by dariusc93** [n0-computer/iroh#385] [Q&A]
Hey! First i would like to say keep up the great work on iroh. I have been keeping my eyes on the project for awhile and been impressed with the progressed made, however do have a question regarding it: Will iroh main focus be towards cli for cloud usage or will one be able use iroh as a library to build their own node into their application as well (apart from iroh-mobile)?

**Comment by b5** [comment]
Hi @dariusc93, apologies for the delay in responding. We've been planning to make iroh work as a library from day 1. We have lots of work to do, but for now the `iroh` crate that targets the CLI binary is set up to consume `iroh-api` as a library, providing both our command line interface and initial example of how to use iroh itself as a library.

The missing piece is being able to embed iroh _services_ (gateway, p2p, store) in a rust binary. We're planning to call that iroh embed, and I'll write up a discussion thread on that soon.

**Reply by conanchen** [reply]
I want to have the api to make IPFS as a normal file system for application developers. https://proto.school/regular-files-api/05 , https://proto.school/mutable-file-system/01

**Reply by conanchen** [reply]
and will  iroh-one work with   ipfsClient from 'ipfs-http-client' ?
